bundles/org.openhab.binding.amberelectric/src/main/java/org/openhab/binding/amberelectric/internal/AmberElectricHandler.java

```diff
@@ -0,0 +1,169 @@
+import org.openhab.core.library.unit.CurrencyUnits;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link AmberElectricHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Paul Smedley - Initial contribution
+ */
+@NonNullByDefault
+public class AmberElectricHandler extends BaseThingHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(AmberElectricHandler.class);
+
+    private long refreshInterval;
+    private String apiKey = "";
+    private String nmi = "";
+    private String siteID = "";
+
+    private @NonNullByDefault({}) AmberElectricConfiguration config;
+    private @NonNullByDefault({}) AmberElectricWebTargets webTargets;
+    private @Nullable ScheduledFuture<?> pollFuture;
+
+    public AmberElectricHandler(Thing thing) {
+        super(thing);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        logger.warn("This binding is read only");
+    }
+
+    @Override
+    public void initialize() {
+        config = getConfigAs(AmberElectricConfiguration.class);
+        if (config.apiKey.isBlank()) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR,
+                    "@text/offline.conf-error.no-api-key");
+            return;
+        }
+
+        webTargets = new AmberElectricWebTargets();
+        updateStatus(ThingStatus.UNKNOWN);
+        refreshInterval = config.refresh;
+        nmi = config.nmi;
+        apiKey = config.apiKey;
+
+        schedulePoll();
+    }
+
+    @Override
+    public void dispose() {
+        super.dispose();
+        stopPoll();
+    }
+
+    private void schedulePoll() {
+        logger.debug("Scheduling poll every {} s", refreshInterval);
+        this.pollFuture = scheduler.scheduleWithFixedDelay(this::poll, 0, refreshInterval, TimeUnit.SECONDS);
+    }
+
+    private void poll() {
+        try {
+            logger.debug("Polling for state");
+            pollStatus();
+        } catch (IOException e) {
+            logger.debug("Could not connect to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        } catch (RuntimeException e) {
+            logger.warn("Unexpected error connecting to AmberAPI", e);
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+    }
+
+    private void stopPoll() {
+        final Future<?> future = pollFuture;
+        if (future != null) {
+            future.cancel(true);
+            pollFuture = null;
+        }
+    }
+
+    private void pollStatus() throws IOException {
+
```

Comment:
Remove this empty line to fix the [checkstyle warning](https://ci.openhab.org/job/PR-openHAB-Addons/21365/Static_20Code_20Analysis_20Report/):
```suggestion
```

Reply:
Done :)